specs/Picker-API_spec.md

```diff
@@ -0,0 +1,121 @@
+   is a less than ideal experience for application developers.
+
+**3. Unable to specify an arbitrary start location for the picker.**
+
+   The existing File/Folder pickers do not allow the developer to specify an arbitrary start location (even if the app has access). Currently, the start location is limited to a set of predefined enum values.
+
+The Reunion picker dialog addresses the above issues. This Reunion API will be based upon the below
+APIs in the Windows SDK. The Reunion Picker API will supercede the existing APIs and add additional methods
+for the functionality: File and folder picker, Multiple folder picker.
+
+- Windows.Storage.Pickers.FileOpenPicker (https://docs.microsoft.com/en-us/uwp/api/Windows.Storage.Pickers.FileOpenPicker)
+- Windows.Storage.Pickers.FileSavePicker (https://docs.microsoft.com/en-us/uwp/api/windows.storage.pickers.filesavepicker)
+- Windows.Storage.Pickers.FolderPicker (https://docs.microsoft.com/en-us/uwp/api/windows.storage.pickers.folderpicker) 
+
+# Description
+
+The goal of this API is to address the gaps in the existing File/Folder picker APIs. The API surface is similar to the existing
+Picker APIs in Windows SDK with additional methods to support File + Folder picker and multiple folder picker. The Project Reunion APIs will supercede the Windows APIs.
+
+# Examples
+## Show file and folder picker
+```c#
+    var picker = new StorageItemPicker();
+    picker.FiltersAndDescriptions.Insert("Images", "*.jpg, *.png");
+    picker.StartLocation = startLocation;
+    picker.PickerKinds = StorageItemPickerKinds.Folder | StorageItemPickerKinds.FileOpen;
+    // multi-select will be automatically set
+    IReadOnlyList<Windows.Storage.IStorageItem> storageItems = await picker.ShowPickerAsync();
+
+    if(0 < storageItems.Count)
+    {
+    }
+```
+## Show single folder picker
+```c#
+    var picker = new StorageItemPicker();
+    picker.PickerKinds = StorageItemPickerKinds.Folder;
+    IReadOnlyList<Windows.Storage.IStorageItem> folder = await picker.ShowPickerAsync();
+```
+## Show multi folder picker
+```c#
+    var picker = new StorageItemPicker();
+    picker.PickerKinds = StorageItemPickerKinds.Folder;
+    picker.MultiSelect = true;
+    IReadOnlyList<Windows.Storage.IStorageItem> folders = await picker.ShowPickerAsync();
+```
+## Show multiple file picker
+```c#
+    var picker = new StorageItemPicker();
+    picker.StartLocation = startLocation;
+    picker.MultuSelect = true;
+    picker.PickerKinds = StorageItemPickerKinds.FileOpen;
+    IReadOnlyList<Windows.Storage.IStorageItem> files = await picker.ShowPickerAsync();
+```
+## Show File Save picker
+```c#
+    var picker = new StorageItemPicker();
+    picker.PickerKinds = StorageItemPickerKinds.FileSave;
+    IReadOnlyList<Windows.Storage.IStorageItem> fileToSave = await picker.ShowPickerAsync();
+```
+
+# Remarks
+# API Notes
+# API Details
+```c#
+    namespace Microsoft.Reunion.Storage.Pickers
```

Comment:
Not sure this is the right namespace yet.

Reply:
Namespaces shouldn't include 'Reunion' (or ProjectReunion) in their name.

Like discussions have recommended existing namespaces but Microsoft. instead of Windows. e.g. Microsoft.Storage.Pickers

@jonwis [docs\Coding-Guidelines.md](https://github.com/microsoft/ProjectReunion/blob/master/docs/Coding-Guidelines.md) makes no mention of namespace guidance. We should add it (in a new WinRT section)?